Compound_Risk_database.R

```diff
@@ -40,44 +42,182 @@ normfuncpos <- function(df, upperrisk, lowerrisk, col1) {
+  filter(country != "Country Level") %>%
+  filter(country != "")
+
+# Save csv with full acapslist
+write.csv(acapslist, "Indicator_dataset/acaps.csv")
+
+# List of countries with specific hazards
+conflictnams <- acapslist %>%
+  filter(str_detect(acapslist$country, c("conflict|Crisis|crisis|Conflict|Refugees|refugees|
+                                         Migration|migration|violence|violence|Boko Haram"))) %>%
+  filter(risk >= 4) %>%
+  dplyr::select(gaplist)
+
+conflictnams <- unique(conflictnams)
+
+# Food security countries
+foodnams <- acapslist[str_detect(acapslist$country, c("Food|food|famine|famine")), ] %>%
+  filter(risk >= 4) %>%
+  dplyr::select(gaplist)
+
+foodnams <- unique(foodnams)
+
+# Natural hazard countries
+naturalnams <- acapslist[str_detect(acapslist$country, c("Floods|floods|Drought|drought|Cyclone|cyclone|
+                                                          Flooding|flooding|Landslides|landslides|
+                                                          Earthquake|earthquake")), ] %>%
+  filter(risk >= 3) %>%
+  dplyr::select(gaplist)
+
+naturalnams <- unique(naturalnams)
+
+# Epidemic countries
+healthnams <- acapslist[str_detect(acapslist$country, c("Epidemic|epidemic")), ] %>%
+  filter(risk >= 3) %>%
+  dplyr::select(gaplist)
+
+healthnams <- unique(healthnams)
+
+# Load countries in the CRM
+countrylist <- read.csv("https://raw.githubusercontent.com/ljonestz/compoundriskdata/master/Indicator_dataset/countrylist.csv")
+
+acapssheet <- countrylist %>%
+  dplyr::select(-X) %>%
+  mutate(
+    Fr_conflict_acaps = case_when(
+      Country %in% unlist(as.list(conflictnams)) ~ 10,
+      TRUE ~ 0
+    ),
+    H_health_acaps = case_when(
+      Country %in% unlist(as.list(healthnams)) ~ 10,
+      TRUE ~ 0
+    ),
+    NH_natural_acaps = case_when(
+      Country %in% unlist(as.list(naturalnams)) ~ 10,
+      TRUE ~ 0
+    ),
+    F_food_acaps = case_when(
+      Country %in% unlist(as.list(foodnams)) ~ 10,
+      TRUE ~ 0
+    )
+  )
+
+# Write ACAPS sheet
+write.csv(acapssheet, "Risk_sheets/acapssheet.csv")
+
+print("ACAPS sheet written")
+
+#
+##
+### ********************************************************************************************
+####    HEALTH: CREATE HEALTH SHEET USING A RANGE OF SOURCE INDICATORS ----
+### ********************************************************************************************
+##
+#
+
+#--------------------—HIS Score-----------------
 HIS <- read.csv("https://raw.githubusercontent.com/ljonestz/compoundriskdata/master/Indicator_dataset/HIS.csv")
 
 HIS <- HIS %>%
   rename(Country = H_Country) %>%
-  select(-X)
+  dplyr::select(-X)
 
 #Normalise scores
 HIS <- normfuncneg(HIS, 20, 70, "H_HIS_Score")
 
-#-----------------------Oxford rollback Score-----------------
-OXrollback <- read.csv("https://raw.githubusercontent.com/OxCGRT/covid-policy-scratchpad/master/rollback_checklist/rollback_checklist.csv")
+#-----------------------—Oxford rollback Score-----------------
```

Comment:
Oxford CGRT switched their dataset from Oxford Rollback Index to Oxford Risk of Openness Index. These changes are to read the new data source. The new source includes current and historical data, so we must select only the most recent data.